components/inputs/input-number.js

```diff
@@ -99,42 +92,84 @@ class InputNumber extends SkeletonMixin(FormElementMixin(LocalizeCoreElement(Lit
 		this.maxExclusive = false;
 		this.minExclusive = false;
 		this.required = false;
-		this.trailingZeroes = false;
 
 		this._formattedValue = '';
 		this._inputId = getUniqueId();
-		this._numDecimalDigits = 0;
+		this._trailingZeroes = false;
+		this._valueTrailingZeroes = '';
+	}
+
+	get maxFractionDigits() {
+		return this._maxFractionDigits ?? Math.max(this.minFractionDigits, 3);
+	}
+	set maxFractionDigits(val) {
+		if (isNaN(val) || val < 0 || val > 20 || val < this.minFractionDigits) {
+			throw new RangeError('maxFractionDigits must be between 0 and 20 and >= minFractionDigits');
+		}
+		this._maxFractionDigits = val;
+		this._updateFormattedValue();
```

Comment:
Not calling `requestUpdate()` because changing min/max fraction digits on its own isn't the thing that should trigger a re-render. It's if the formatted value changes that should.